### PR TITLE
Continuous deployment

### PR DESCRIPTION
**Pull request by YashoSharma**
- [x] Adds deployment hooks

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,10 +12,26 @@ addons:
...
+  secret_access_key:
+    secure: "bCSmJe3dcBMDwDXzK7HWhkSJPNOrHDbv6uAKv8yD/Ny6Qo+EfJdw81ygGvKBHepI+dj3fPnDAZZBUruRzzLqMFMMzzQv/cunZhM+wFyjTZP4KP2iednk1MANdzivkfJj/BoyfUD+c5OOr5OK83zHgQomsv35XnwF9v7TCbz59fN/jwN15fU+Y161rtSoU/8JutWAQ2oQzXbRkXu9zr1tPkeT4wjjl/+wCfplLUWocwQx2OA+VJI66tTuURzwtLtnZ8zLDbyukY+kG41b8dRdx5agdD4CT8QytSst527zVxualYK3Mhjp4T79CDXZMOfn53xPK2H3gklnca+H+mjple5HS/i6aItfYoqOqCqFf2EiJ7rFliNP0tCvf3eKJIsGPPvMaqoRRep7ZJmyTDvIg9WyjqckXtYY/ck1FuvwdcibbEpvQD/NrrkFjVhTukxbdMhIIjYg1FkPic1q5d0AOm8txgWhKgu1tP1tzyOK/4qmC3PtVG4kKH+J1P7pzztxPTFEh5CNvdbfI/Qt+94QncFu0vgrI9vwnch24Jz7t+zhVCmLCgF1FYp915SisaTlBN+RQ0sdI/0kbd58Mb7co87psPVULPhVEgz66L/TJSXIfGWXfteRpNqCyH6upKICfg7b7Uy7M0M5oCMn8QiJyA5LBqOtbVvpkYlwlzjPcZE="
```
Comment (redsn0w422) [inline .travis.yml]: What are these keys for?

Comment (YashoSharma) [inline .travis.yml]: They're the access keys for AWS, but encrypted with travis 
https://docs.travis-ci.com/user/encryption-keys/

Comment (redsn0w422) [inline .travis.yml]: "Encrypted" meaning no one else can use them? Just want to make sure, since this is open source :)

Comment (YashoSharma) [inline .travis.yml]: They'd need the public key associated with this repo on travis to decrypt them

File: .travis.yml [hunk 2]
```diff
@@ -12,10 +12,25 @@ addons:
 cache:
   directories:
     - node_modules
-services:
-  #- docker
 before_script:
-  #Docker setup goes here
   - pwd
   - cp ~/build/HackIllinois/api-2017/config/test.config.template ~/build/HackIllinois/api-2017/config/test.config
```
Comment (nmagerko) [inline .travis.yml]: Is this line necessary if we're deploying to production? I see a production config is getting copied over at the end, but would that mean we have both a prod and a test config installed?

File: .travis.yml (hunk 1 above)
Comment (nmagerko) [inline .travis.yml]: This is cool. I didn't know they offered this. Nice find.

File: .travis.yml [hunk 3]
```diff
@@ -12,10 +12,25 @@ addons:
...
+  region: "us-east-2"
+  app: "hackillinois-api"
+  env: "api-hackillinois"
+  bucket_name:
+    secure: "zU+i6unLI5R8q+L3ZqqJupJmQlfkH4IEvNBaM+mX+jgKqMHZ808sZhlTV/urKvoTlQ/s5V2Z1UfK9piPE6lhzpYq4L5/A9LG5rudxcKGnijU5emvhRISrJYM/e4XtMdFQ/9faHsSayFObj+qvR18zExjRXei1ycxsB3r9KRkR/t5LI4DfhnRZtZYgu4fOkDxPcGLSc5R3/hg7ZUaGquK5llz2Lq/GvmRKBZXlkYyuSg+jMda3dnNkF7gr6K4IwNFq3hGzrBkhOAL5D30q4oewGKqjUsidMtQPPFKXLN0jshHnhme5al/gqMHt8QhWKzVR02VHxoGh+0pyNf2IcqcmsEFTN9O7/OCrLpYFsZmyHIc22E7quWawSa1Hi8xLSKrMOf/IHcPm5z9XAcZr85x3Ed/DPFkZDqiWzGMsQBphCZ3WkNcVn/aRoCCvozZgvstyi5joc8GXFM/YMPvYsQ1ol0BrLYVSqqvXJ6hZF3J38aUDegHwIS9oh5K/23XCz5d0wSf3jYVNdY4kCueTYLgoMSf3fNUYnJx4SLdLUDWpgTm+sN7IzLG09WAWAPqmDCTMT28U8nkA7nzAJPg/ZPHWDodiJRe9TopERiBbtnASxp5cXb614LfpUmfnNKilqRXvT559itHnohft9tT+sXneI70vZygrKtctDonWz+Bl4Y="
```
Comment (nmagerko) [inline .travis.yml]: Thank you for encrypting this too 👍

**Comment by nmagerko** [comment]
Were you planning to do a release after merging this, or will that need to wait until after the tests are updated?

File: .travis.yml (hunk 2 above)
Comment (YashoSharma) [inline .travis.yml]: `before_script` runs before travis runs the unit tests which need the test env, and the production config should be copied over within the s3 bucket that the api is being deployed to

Comment (nmagerko) [inline .travis.yml]: Nice. Didn't know that.

**Comment by YashoSharma** [comment]
The tests should be close to completion so I'll release after those are merged into staging.

**Pull request merged**